google/cloud/pubsub/internal/subscription_session_test.cc

```diff
@@ -305,6 +306,130 @@ TEST(SubscriptionSessionTest, ScheduleCallbacksExactlyOnce) {
   for (auto& t : tasks) t.join();
 }
 
+/// @test Verify callbacks are scheduled in the background threads with Open
+/// Telemetry enabled.
+TEST(SubscriptionSessionTest, ScheduleCallbacksWithOtelEnabled) {
+  auto mock = std::make_shared<pubsub_testing::MockSubscriberStub>();
+  pubsub::Subscription const subscription("test-project", "test-subscription");
+
+  std::mutex mu;
+  int count = 0;
+  std::mutex ack_id_mu;
+  std::condition_variable ack_id_cv;
+  int expected_ack_id = 0;
+  auto constexpr kAckCount = 100;
+
+  google::cloud::CompletionQueue cq;
+  using TimerFuture = future<StatusOr<std::chrono::system_clock::time_point>>;
+  using us = std::chrono::microseconds;
+  EXPECT_CALL(*mock, AsyncAcknowledge)
+      .WillRepeatedly(
+          [&](google::cloud::CompletionQueue&, auto, auto,
+              google::pubsub::v1::AcknowledgeRequest const& request) {
+            for (auto const& a : request.ack_ids()) {
+              std::lock_guard<std::mutex> lk(ack_id_mu);
+              EXPECT_EQ("test-ack-id-" + std::to_string(expected_ack_id), a);
+              ++expected_ack_id;
+              if (expected_ack_id >= kAckCount) ack_id_cv.notify_one();
+            }
+            return cq.MakeRelativeTimer(us(10)).then(
+                [](TimerFuture) { return Status{}; });
+          });
+  EXPECT_CALL(*mock, AsyncModifyAckDeadline)
+      .WillRepeatedly([](google::cloud::CompletionQueue&, auto, auto,
+                         google::pubsub::v1::ModifyAckDeadlineRequest const&) {
+        return make_ready_future(Status{});
+      });
+  EXPECT_CALL(*mock, AsyncStreamingPull)
+      .Times(AtLeast(1))
+      .WillRepeatedly([&](google::cloud::CompletionQueue const&, auto, auto) {
+        auto stream = std::make_unique<pubsub_testing::MockAsyncPullStream>();
+        EXPECT_CALL(*stream, Start).WillOnce([&] {
+          return cq.MakeRelativeTimer(us(10)).then(
+              [](TimerFuture) { return true; });
+        });
+        EXPECT_CALL(*stream, Write)
+            .WillOnce(
+                [&](google::pubsub::v1::StreamingPullRequest const& request,
+                    grpc::WriteOptions const&) {
+                  EXPECT_EQ(subscription.FullName(), request.subscription());
```

Comment:
in keeping with the idea of making tests as simple as possible, I do not set expectations like this. That way the expectations that actually matter stand out. Some people disagree with me.

Reply:
ack, they already existed as this. i think the refactoring makes that clearer.